app.js

```diff
@@ -148,6 +160,66 @@ app.get('/macros/:macro.json', function (req, res) {
   }
 });
 
+// List all devices in JSON format
+app.get('/devices.json', function(req, res) {
+    // TODO: Should return a nicely formatted response, not just the config object
+    res.json(config.devices);
+});
+
+// List all commands for :device in JSON format
+app.get('/devices/:device.json', function(req, res) {
+    if (config.devices && config.devices[req.params.device]) {
+        // TODO: Should return a nicely formatted response, not just the config object
+        res.json(config.devices[req.params.device]);
+    } else {
+        res.send(404);
+    }
+});
+
+// Get device state
+app.get('/devices/:device', function(req, res) {
+    if (config.devices &&
+        config.devices[req.params.device] &&
+        config.devices[req.params.device]['state']) {
+
+        var state = config.devices[req.params.device]['state'];
+
+        // Build request to make to get device state
+        var stateReq = {
+          method: state.method || "GET",
+          url: state.url,
+          form: state.body
+        };
+
+        // Make request for state, include body of response in response
+        request(stateReq, function (error, response, body) {
+          // TODO: How to parse the response and only return relevant portion?
+          var bodyJson = JSON.parse(body);
+          res.json(bodyJson);
+        });
+
+    } else {
+        // Device doesn't have a state, return 404
+        res.send(404);
+    }
+});
+
+// Execute device action
+app.post('/devices/:device/:command', function(req, res) {
```

Comment:
Yeah, I think that a 'device' should represent a physical artifact within the region the universal remote is responsible for controlling. A device could use any number of protocols but would be presented in a 'unified' way for the UI / API to interact with.

I hadn't considered a device that uses different protocols to get state vs send commands from, but it seems like something that could be supported.